### PR TITLE
CustomViewPart06! Finish!

### PR DESCRIPTION
**Pull request by at-bienle**
Hehe!

### CODE REVIEW

File: CustomView/CustomViewPart06/CustomViewPart06/HomeViewController.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        if let slideTemp = sliderUI {
+            slideTemp.delegate = self
+            slideTemp.frame = CGRect(x: 170, y: 170, width: 50, height: 360)
+            view.addSubview(slideTemp)
+        }
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+    }
```
Comment (at-truongnguyen) [inline CustomView/CustomViewPart06/CustomViewPart06/HomeViewController.swift]: please remove unused code

File: CustomView/CustomViewPart06/CustomViewPart06/HomeViewController.swift [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
```
Comment (at-truongnguyen) [inline CustomView/CustomViewPart06/CustomViewPart06/HomeViewController.swift]: please add `super`

File: CustomView/CustomViewPart06/CustomViewPart06/HomeViewController.swift [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        self.view.endEditing(true)
```
Comment (at-truongnguyen) [inline CustomView/CustomViewPart06/CustomViewPart06/HomeViewController.swift]: remove self please

File: CustomView/CustomViewPart06/CustomViewPart06/HomeViewController.swift [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @IBAction func whenRditPercent(_ sender: Any) {
```
Comment (at-truongnguyen) [inline CustomView/CustomViewPart06/CustomViewPart06/HomeViewController.swift]: Sorry I don't understand that func name.

File: CustomView/CustomViewPart06/CustomViewPart06/HomeViewController.swift [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        if let percent = percentNumber.text {
+            if let numberPercent = Double(percent) {
```
Comment (at-truongnguyen) [inline CustomView/CustomViewPart06/CustomViewPart06/HomeViewController.swift]: You should merge these code.

File: CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+//  SliderView.swift
+//  CustomViewPart06
+//
+//  Created by AST on 6/16/17.
+//  Copyright © 2017 AST. All rights reserved.
+//
+
+import UIKit
+protocol SliderViewDelegrate: class {
+    func changePercent(view: SliderView)
```
Comment (at-truongnguyen) [inline CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift]: Your func name is not correct, could you ask someone for help?

File: CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+        //sliderView.layer.borderColor = UIColor(red: 79 / 255, green: 129 / 255, blue: 173 / 255, alpha: 1.0).cgColor
+        //sliderView.layer.backgroundColor = UIColor(red: 102 / 255, green: 165 / 255, blue: 218 / 255, alpha: 1.0).cgColor
+
+        //tempButton.layer.borderWidth = 1
+        //tempButton.layer.borderColor = UIColor(red: 189 / 255, green: 120 / 255, blue: 64 / 255, alpha: 1.0).cgColor
+        //tempButton.layer.backgroundColor = UIColor(red: 239 / 255, green: 137 / 255, blue: 56 / 255, alpha: 1.0).cgColor
+        tempButton.layer.cornerRadius = 60
+        //tempButton.layer.masksToBounds = true
+        tempButton.isUserInteractionEnabled = false
+        tempButton.center = CGPoint(x: (sliderView.frame.origin.x + sliderView.frame.width / 2), y: (sliderView.frame.origin.y + sliderView.frame.height / 2))
```
Comment (at-truongnguyen) [inline CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift]: Why don't you use `center` for reducing code?

File: CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        //sliderView.layer.borderWidth = 1
+        //sliderView.layer.borderColor = UIColor(red: 79 / 255, green: 129 / 255, blue: 173 / 255, alpha: 1.0).cgColor
+        //sliderView.layer.backgroundColor = UIColor(red: 102 / 255, green: 165 / 255, blue: 218 / 255, alpha: 1.0).cgColor
+
+        //tempButton.layer.borderWidth = 1
+        //tempButton.layer.borderColor = UIColor(red: 189 / 255, green: 120 / 255, blue: 64 / 255, alpha: 1.0).cgColor
+        //tempButton.layer.backgroundColor = UIColor(red: 239 / 255, green: 137 / 255, blue: 56 / 255, alpha: 1.0).cgColor
```
Comment (at-truongnguyen) [inline CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift]: Please delete all unused code.

File: CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+//
+
+import UIKit
+protocol SliderViewDelegrate: class {
+    func changePercent(view: SliderView)
+}
+
+class SliderView: UIView {
+
+    @IBOutlet weak var tempButton: UIButton!
```
Comment (at-truongnguyen) [inline CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift]: COuld you explain why it just `temp`?

File: CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    
+    override func touchesMoved(_ touches: Set<UITouch>, with event: UIEvent?) {
+        super.touchesMoved(touches, with: event)
+        guard let loc = touches.first?.location(in: self) else { return }
+        if isTouchOnButton {
+            if (loc.y <= (sliderView.frame.origin.y + sliderView.frame.height)) && (loc.y >= sliderView.frame.origin.y) {
+                tempButton.center.y = loc.y
+                let x: Int = Int (100 - (tempButton.center.y - sliderView.frame.origin.y) * 100 / (sliderView.frame.height))
```
Comment (at-truongnguyen) [inline CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift]: `x` is percent, isn't it? Please correct your variable

File: CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+
+import UIKit
+protocol SliderViewDelegrate: class {
+    func changePercent(view: SliderView)
+}
+
+class SliderView: UIView {
+
+    @IBOutlet weak var tempButton: UIButton!
+    @IBOutlet weak var sliderView: UIView!
```
Comment (at-truongnguyen) [inline CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift]: please make it private

File: CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+    @IBOutlet weak var blueView: UIView!
```
Comment (at-truongnguyen) [inline CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift]: please make it private

File: CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        //sliderView.layer.borderWidth = 1
+        //sliderView.layer.borderColor = UIColor(red: 79 / 255, green: 129 / 255, blue: 173 / 255, alpha: 1.0).cgColor
+        //sliderView.layer.backgroundColor = UIColor(red: 102 / 255, green: 165 / 255, blue: 218 / 255, alpha: 1.0).cgColor
+
+        //tempButton.layer.borderWidth = 1
+        //tempButton.layer.borderColor = UIColor(red: 189 / 255, green: 120 / 255, blue: 64 / 255, alpha: 1.0).cgColor
+        //tempButton.layer.backgroundColor = UIColor(red: 239 / 255, green: 137 / 255, blue: 56 / 255, alpha: 1.0).cgColor
+        tempButton.layer.cornerRadius = 60
```
Comment (at-truongnguyen) [inline CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift]: Could you tell me why `cornerRadius = 60` while that button's width = height = 60 also?

File: CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift (hunk 9 above)
Comment (at-bienle) [inline CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift]: :D name's button must be percentButton 🗡

File: CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift (hunk 10 above)
Comment (at-bienle) [inline CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift]: x is the number must be in title of button, it be used to get number so i thought x may be correct

File: CustomView/CustomViewPart06/CustomViewPart06/HomeViewController.swift (hunk 1 above)
Comment (at-bienle) [inline CustomView/CustomViewPart06/CustomViewPart06/HomeViewController.swift]: done

File: CustomView/CustomViewPart06/CustomViewPart06/HomeViewController.swift (hunk 2 above)
Comment (at-bienle) [inline CustomView/CustomViewPart06/CustomViewPart06/HomeViewController.swift]: done

File: CustomView/CustomViewPart06/CustomViewPart06/HomeViewController.swift (hunk 3 above)
Comment (at-bienle) [inline CustomView/CustomViewPart06/CustomViewPart06/HomeViewController.swift]: done

File: CustomView/CustomViewPart06/CustomViewPart06/HomeViewController.swift (hunk 4 above)
Comment (at-bienle) [inline CustomView/CustomViewPart06/CustomViewPart06/HomeViewController.swift]: it's Edit ✌️

File: CustomView/CustomViewPart06/CustomViewPart06/HomeViewController.swift (hunk 5 above)
Comment (at-bienle) [inline CustomView/CustomViewPart06/CustomViewPart06/HomeViewController.swift]: done

File: CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift (hunk 7 above)
Comment (at-bienle) [inline CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift]: done

File: CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift (hunk 8 above)
Comment (at-bienle) [inline CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift]: done

File: CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift (hunk 11 above)
Comment (at-bienle) [inline CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift]: done

File: CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift (hunk 12 above)
Comment (at-bienle) [inline CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift]: done

File: CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift (hunk 13 above)
Comment (at-bienle) [inline CustomView/CustomViewPart06/CustomViewPart06/SliderView.swift]: :D i test number of radius until is same same circle so have that :D

**Pull request merged**